### PR TITLE
Use GH OIDC for deployments

### PR DESCRIPTION
**Pull request by zaro0508**
Use the more secure GH OIDC method for deploying to AWS

depends on https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/907

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -64,3 +66,4 @@ jobs:
       job-environment: "org-sagebase-strides-ampad-workflows"
       sceptre-suffix: "ampad"
       tower-url: "https://tower.sagebionetworks.org"
+      aws-role-to-assume: "arn:aws:iam::751556145034:role/sagebase-github-oidc-nextflow-infra"
```
Comment (zaro0508) [inline .github/workflows/main.yaml]: the role name is generated by AWS, will need deploy PR https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/902 then get the name and put it here.

**Review by thomasyu888 — APPROVED** [review]
LGTM!

File: .github/workflows/main.yaml (hunk 1 above)
Comment (thomasyu888) [reply .github/workflows/main.yaml]: @zaro0508 is this done?

Comment (zaro0508) [reply .github/workflows/main.yaml]: ok, it's done now.

**Pull request merged**